processing/src/main/java/org/apache/carbondata/processing/loading/parser/impl/RowParserImpl.java

```diff
@@ -34,8 +37,12 @@
   private int numberOfColumns;
 
   public RowParserImpl(DataField[] output, CarbonDataLoadConfiguration configuration) {
-    String[] complexDelimiters =
+    String[] tempComplexDelimiters =
         (String[]) configuration.getDataLoadProperty(DataLoadProcessorConstants.COMPLEX_DELIMITERS);
+    Queue<String> complexDelimiters = new LinkedList<>();
+    for (int i = 0; i < 4; i++) {
```

Comment:
Avoid using hardcoded values like `4` in the loops while coding....instead iterate over tempComplexDelimiters length

Reply:
done